File: урок2/sumOfThree.js
const readline = require('readline').createInterface({
    input: process.stdin,
    output: process.stdout
});

let input = [];
readline.on('line', (line) => {
    input.push(line);
    if (input.length === 2) {
        readline.close();
    }
}).on('close', () => {
    const n = parseInt(input[0]);
    const arr = input[1].split(' ').map(Number);
    const result = calculateResult(n, arr);
    console.log(result);
});

function calculateResult(n, arr) {
    const mod = 10**9 + 7;
    const prefSums = Array(n + 1).fill(0);
    for (let i = 1; i <= n; i++) {
        prefSums[i] = (arr[i - 1] + prefSums[i - 1]) % mod;
    }

    const suffSums = Array(n + 1).fill(0);
    for (let i = n - 1; i >= 0; i--) {
        suffSums[i] = (suffSums[i + 1] + arr[i]) % mod;
    }

    let ans = 0;
    for (let nowPos = 1; nowPos < n - 1; nowPos++) {
        ans = (ans + prefSums[nowPos] * arr[nowPos] * suffSums[nowPos + 1]) % mod;
    }

    return ans;
}

File: урок2/sumOfThree.py
n = int(input())
arr = list(map(int, input().split()))
mod = 10**9 + 7
pref_sums = [0] * (n + 1)
for i in range(1, n + 1):
    pref_sums[i] = (pref_sums[i-1] + arr[i-1]) % mod
suff_sums = [0] * (n+1)
for i in range (n - 1, -1, -1):
    suff_sums[i] = (suff_sums[i+1] + arr[i]) % mod
ans = 0
for now_pos in range(1, n - 1):
    ans = (ans + pref_sums[now_pos] * arr[now_pos] * suff_sums[now_pos + 1]) % mod
print(ans)




""" n = int(input())
arr = list(map(int, input().split()))
mod = 10**9 + 7
pref_sums = [0]*(n+1)
for i in range(1, n + 1):
    pref_sums[i] = (arr[i-1] + pref_sums[i-1]) % mod
suff_sums = [0] * (n+1)
for i in range(n-1,-1, -1):
    suff_sums[i] = (suff_sums[i+1] + arr[i] % mod)
ans = 0
for now_pos in range (1, n-1):
    ans = (ans + pref_sums[now_pos] * arr[now_pos] * suff_sums[now_pos + 1]) % mod
print(ans) """